src/palace/manager/data_layer/bibliographic.py

```diff
@@ -770,25 +772,48 @@ def update_contributions(
                     viaf=contributor_data.viaf,
                 )
                 new_contributors.append(contributor.id)
-                if contributor_data.display_name:
+                if (
+                    contributor_data.display_name
+                    and contributor_data.display_name != contributor.display_name
+                ):
                     contributor.display_name = contributor_data.display_name
-                if contributor_data.biography:
+                if (
+                    contributor_data.biography
+                    and contributor_data.biography != contributor.biography
+                ):
                     contributor.biography = contributor_data.biography
-                if contributor_data.aliases:
+                if (
+                    contributor_data.aliases
+                    and contributor_data.aliases != contributor.aliases
+                ):
                     contributor.aliases = contributor_data.aliases
-                if contributor_data.lc:
+                if contributor_data.lc and contributor_data.lc != contributor.lc:
                     contributor.lc = contributor_data.lc
-                if contributor_data.viaf:
+                if contributor_data.viaf and contributor_data.viaf != contributor.viaf:
                     contributor.viaf = contributor_data.viaf
-                if contributor_data.wikipedia_name:
+                if (
+                    contributor_data.wikipedia_name
+                    and contributor_data.wikipedia_name != contributor.wikipedia_name
+                ):
                     contributor.wikipedia_name = contributor_data.wikipedia_name
             else:
                 self.log.info(
                     "Not registering %s because no sort name, LC, or VIAF",
                     contributor_data.display_name,
                 )
 
-        if sorted(old_contributors) != sorted(new_contributors):
-            contributors_changed = True
+        old_contributors_set = set(old_contributors)
+        new_contributors_set = set(new_contributors)
+        contributors_changed = old_contributors_set != new_contributors_set
+        if contributors_changed and replace:
```

Comment:
Same approach as above to prevent insertions and deletions when no change in the data has occurred.